vscode-extension/editor/src/VSCodeEditor.tsx

```diff
@@ -65,23 +67,27 @@ export default function VSCodeEditor() {
 
   const { classes } = useStyles();
 
-  const updateContent = useCallback(async (text: string) => {
-    // TODO: saqadri - this won't work for YAML -- the handling of the text needs to include the logic from AIConfig.load
+  const updateContent = useCallback(
+    async (text: string) => {
+      // TODO: saqadri - this won't work for YAML -- the handling of the text needs to include the logic from AIConfig.load
+
+      if (text != null) {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        let updatedConfig: any = {};
+        try {
+          // Try parsing the string as JSON first
+          updatedConfig = JSON.parse(text);
+        } catch (e) {
+          // If that fails, try parsing the string as YAML
+          updatedConfig = yaml.load(text);
+        }
 
-    if (text != null) {
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      let updatedConfig: any = {};
-      try {
-        // Try parsing the string as JSON first
-        updatedConfig = JSON.parse(text);
-      } catch (e) {
-        // If that fails, try parsing the string as YAML
-        updatedConfig = yaml.load(text);
+        setAIConfig(updatedConfig);
+        updateWebviewState(vscode, { aiconfigState: updatedConfig });
```

Comment:
This and adding vscode to deps array is the only change to this callback. The rest is auto-formatting